modelci/hub/profile_.py

```diff
@@ -0,0 +1,125 @@
+import os
+import random
+import time
+
+import docker
+from modelci.hub.client import CVTFSClient, CVTorchClient, CVONNXClient, CVTRTClient
+from modelci.metrics.benchmark.metric import BaseModelInspector
+from modelci.persistence.exceptions import ServiceException
+from modelci.types.models.mlmodel import MLModel, Engine
+from modelci.types.models.profile_results import DynamicProfileResult, ProfileMemory, ProfileThroughput, ProfileLatency
+from modelci.utils import Logger
+from modelci.utils.misc import get_ip
+from modelci.hub.deployer.dispatcher import serve
+DEFAULT_BATCH_NUM = 100
+
+random.seed(ord(os.urandom(1)))
+logger = Logger(__name__, welcome=False)
+
+
+class Profiler(object):
+    """Profiler class, call this to test model performance.
+
+    Args:
+        model_info (MLModel): Information about the model, can get from `retrieve_model` method.
+        inspector (BaseModelInspector): The client instance implemented from :class:`BaseModelInspector`.
+    """
+
+    def __init__(self, model_info: MLModel, inspector: BaseModelInspector = None):
+        """Init a profiler object."""
+        self.model = model_info
+        self.docker_client = docker.from_env()
+        if inspector is None:
+            self.inspector = self.__auto_select_client()  # TODO: To Improve
+        else:
+            if isinstance(inspector, BaseModelInspector):
+                self.inspector = inspector
+            else:
+                raise TypeError("The inspector should be an instance of class BaseModelInspector!")
+
+    def pre_deploy(self, device='cuda'):
```

Comment:
Why this name is pre-deploy? In my point of view, you start the docker container here, so it should be a deploy function?

Reply:
Yes. In the functional aspect, this `pre_deploy`  essentially dose a deploy serve job. While in the aspect of logical, it has to be a 'pre' deploy. Because according to the workflow of MLModel-CI, profile are supposed to be carried out before deployment.